src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -71,13 +72,28 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
                 }
 
                 bool success = _txFilterPipeline.Execute(tx, parent);
-                if (success)
+                if (!success) continue;
+
+                if (tx.Type == TxType.Blob)
                 {
-                    if (_logger.IsTrace) _logger.Trace($"Selected {tx.ToShortString()} to be potentially included in block.");
+                    int txAmountOfBlobs = tx.BlobVersionedHashes?.Length ?? 0;
+                    if ((blobsCounter + txAmountOfBlobs) > Eip4844Constants.MaxBlobsPerBlock)
+                    {
+                        if (_logger.IsTrace) _logger.Trace($"Declining {tx.ToShortString()}, no more blob space.");
+                        continue;
+                    }
 
-                    selectedTransactions++;
-                    yield return tx;
+                    blobsCounter += txAmountOfBlobs;
+                    if (_logger.IsTrace) _logger.Trace($"Selected shard blob tx {tx.ToShortString()} to be potentially included in block, total blobs included: {blobsCounter}.");
                 }
+                else
+                {
+                    if (_logger.IsTrace)
+                        _logger.Trace($"Selected {tx.ToShortString()} to be potentially included in block.");
+                }
+
+                selectedTransactions++;
+                yield return tx;
```

Comment:
Open question - would we have separate TxSource for blobs, or we will mix it with standard pool?

Reply:
Let's think of it with TxPool redesign for blobs.